Index: git/xdma/Makefile
===================================================================
--- git.orig/xdma/Makefile
+++ git/xdma/Makefile
@@ -33,16 +33,11 @@ ifneq ($(KERNELRELEASE),)
 	$(TARGET_MODULE)-objs := libxdma.o xdma_cdev.o cdev_ctrl.o cdev_events.o cdev_sgdma.o cdev_xvc.o cdev_bypass.o xdma_mod.o xdma_thread.o
 	obj-m := $(TARGET_MODULE).o
 else
-	BUILDSYSTEM_DIR:=/lib/modules/$(shell uname -r)/build
+	KERNEL_SRC:=/lib/modules/$(shell uname -r)/build
 	PWD:=$(shell pwd)
-all :
-	$(MAKE) -C $(BUILDSYSTEM_DIR) M=$(PWD) modules
+all: modules
 
-clean:
-	$(MAKE) -C $(BUILDSYSTEM_DIR) M=$(PWD) clean
-	@/bin/rm -f *.ko modules.order *.mod.c *.o *.o.ur-safe .*.o.cmd
-
-install: all
-	$(MAKE) -C $(BUILDSYSTEM_DIR) M=$(PWD) modules_install
+modules modules_install clean:
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) $@
 
 endif
