From 04346b516104b1ae33d4b391b8880ebdae4dc61c Mon Sep 17 00:00:00 2001
From: Oliver Talevski <oliver.talevski@esss.se>
Date: Fri, 3 May 2019 15:22:52 +0200
Subject: [PATCH 3/4] hdf5 threads

---
 configure.ac | 19 +------------------
 1 file changed, 1 insertion(+), 18 deletions(-)

diff --git a/configure.ac b/configure.ac
index 796352f..b8db024 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1680,24 +1680,7 @@ AC_DEFINE_UNQUOTED([PRINTF_LL_WIDTH], ["$hdf5_cv_printf_ll"],
 ## Check if pthread_attr_setscope(&attribute, PTHREAD_SCOPE_SYSTEM)
 ## is supported on this system
 ##
-AC_MSG_CHECKING([Threads support system scope])
-AC_CACHE_VAL([hdf5_cv_system_scope_threads],
-  [AC_TRY_RUN([
-    #if STDC_HEADERS
-    #include <stdlib.h>
-    #include <pthread.h>
-    #endif
-
-    int main(void)
-    {
-      pthread_attr_t attribute;
-      int ret;
-
-      pthread_attr_init(&attribute);
-      ret=pthread_attr_setscope(&attribute, PTHREAD_SCOPE_SYSTEM);
-      exit(ret==0 ? 0 : 1);
-    }
-  ], [hdf5_cv_system_scope_threads=yes], [hdf5_cv_system_scope_threads=no],)])
+hdf5_cv_system_scope_threads=no
 
 if test ${hdf5_cv_system_scope_threads} = "yes"; then
   AC_DEFINE([SYSTEM_SCOPE_THREADS], [1],
-- 
2.7.4

