From 18c1802acb475e19eec93390252f83e8db604c2e Mon Sep 17 00:00:00 2001
From: Oliver Talevski <oliver.talevski@esss.se>
Date: Fri, 3 May 2019 15:24:28 +0200
Subject: [PATCH 4/4] hdf5 shared

---
 configure.ac | 19 ++-----------------
 1 file changed, 2 insertions(+), 17 deletions(-)

diff --git a/configure.ac b/configure.ac
index b8db024..664cc6d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2871,23 +2871,8 @@ touch ./config/stamp2
 ## Finally the makefiles
 test "$no_create" = yes || ${CONFIG_SHELL-/bin/sh} $CONFIG_STATUS || exit 1
 
-## Are we compiling static libraries, shared libraries, or both?  This
-## is only used for the libhdf5.settings file. We can't just look at
-## $enable_static and $enable_shared because if they're yes the ltconfig
-## might have decided that one or the other is simply not possible.
-## Therefore we have to ask the generated `libtool' shell script
-## which 'features' it has enabled.
-if (./libtool --features | grep '^enable shared libraries' > /dev/null); then
-  enable_shared=yes
-else
-  enable_shared=no
-fi
-
-if (./libtool --features | grep '^enable static libraries' > /dev/null); then
-  enable_static=yes
-else
-  enable_static=no
-fi
+enable_shared=yes
+enable_static=no
 
 if test "X$enable_static" = "Xyes" && test "X$enable_shared" = "Xyes"; then
   STATIC_SHARED="static, shared"
-- 
2.7.4

