diff --git a/setup.py b/setup.py
index 7ca751f..7e0d1eb 100644
--- a/setup.py
+++ b/setup.py
@@ -45,8 +45,7 @@ problem before tyring to use the epics package.
 
 nolibca = os.environ.get('NOLIBCA', None)
 if nolibca is None:
-    pkg_data = {'epics.clibs': ['darwin64/*', 'linux64/*', 'linux32/*',
-                                'linuxarm/*', 'win32/*', 'win64/*']}
+    pkg_data = {'epics.clibs': ['linux64/*']}
 else:
     pkg_data = dict()
 
