From 6a0b9a3a157cebfa323ad78b1630b551c4fdbf53 Mon Sep 17 00:00:00 2001
From: Oliver Talevski <oliver.talevski@esss.se>
Date: Tue, 6 Nov 2018 11:49:39 +0100
Subject: [PATCH] Adapt Makefile for Yocto

---
 Makefile | 23 +++++------------------
 1 file changed, 5 insertions(+), 18 deletions(-)

diff --git a/Makefile b/Makefile
index 9b70c53..f91b1b5 100644
--- a/Makefile
+++ b/Makefile
@@ -49,26 +49,13 @@ tsc-objs := tscdrvr.o ioctllib.o tscklib.o maplib.o irqlib.o mapmaslib.o  rdwrli
 	dmalib.o timerlib.o fifolib.o i2clib.o semaphorelib.o
 pon-objs := pondrvr.o
 
-ifeq ($(ARCH),powerpc)
-# JFG platform
-#	KERNELDIR = /T2080IFC_2.0/linux/build/
-# XP platform
-	#KERNELDIR = /data/T2081/rootfs_2080_1.9/lib/modules/3.12.37-rt51+g43cecda/build/
-	#KERNELDIR = /data/QorIQ-SDK-V2.0-20160527-yocto/build_t2080rdb-64b/tmp/work/t2080rdb_64b-fsl-linux/linux-qoriq/4.1-r0/build/
-	#KERNELDIR = /data/QorIQ-SDK-V2.0-20160527-yocto/build_t2080rdb-64b/tmp/work/t2080rdb_64b-fsl-linux/linux-qoriq/4.1-r0/build/
-	KERNELDIR =  /data/QorIQ-SDK-V2.0-20160527-yocto/build_ifc1211/tmp/work/ifc1211-fsl-linux/linux-qoriq/4.1-r0/build/
-else
-	KERNELDIR ?= /lib/modules/$(shell uname -r)/build
-endif
 EXTRA_CFLAGS := $(ADD_CFLAGS)
 PWD := $(shell pwd)
-default:
-ifeq ($(ARCH),powerpc)
-	$(MAKE) -I . ARCH=$(ARCH) CROSS_COMPILE=$(CROSS_COMPILE)  -C $(KERNELDIR) M=$(PWD) modules
-else
-	$(MAKE) -I . -C $(KERNELDIR) M=$(PWD) modules
-endif
+all:
+	$(MAKE) -I . -C $(KERNEL_SRC) M=$(PWD)
+modules_install:
+	$(MAKE) -I . -C $(KERNEL_SRC) M=$(PWD) modules_install
 clean:
-	$(MAKE) -C $(KERNELDIR) M=$(PWD) clean
+	$(MAKE) -C $(KERNEL_SRC) M=$(PWD) clean
 	@rm -f Module.symvers
 	$(RM) *.o *.ko *.mod.c
-- 
2.7.4

