From d93de80f74458bf49e873b2a41d5473a8ab4b031 Mon Sep 17 00:00:00 2001
From: Oliver Talevski <oliver.talevski@esss.se>
Date: Fri, 23 Nov 2018 11:22:14 +0100
Subject: [PATCH] Enable FMCs before exiting

---
 TscMon.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/TscMon.c b/TscMon.c
index 464761c..864c8b5 100644
--- a/TscMon.c
+++ b/TscMon.c
@@ -571,6 +571,9 @@ int main(int argc, char *argv[]){
 			}
 		}
 		else if(argv[2][0] == 'e'){
+			//Enable FMCs and exit. DDR already calibrated
+			data = 0xC0000000;
+			tsc_pon_write(tsc_fd, 0xC, &data);
 			goto TscMon_exit;
 		}
 	}
-- 
2.7.4

