diff --git a/sis8300_ioctl.c b/sis8300_ioctl.c
index 05d6715..287ff4a 100644
--- a/sis8300_ioctl.c
+++ b/sis8300_ioctl.c
@@ -5,7 +5,7 @@
 #include <linux/fs.h>
 #include <linux/wait.h>
 #include <linux/sched.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 
 #include "sis8300.h"
 #include "sis8300_defs.h"
diff --git a/sis8300_mmap.c b/sis8300_mmap.c
index 89ca343..2b02985 100644
--- a/sis8300_mmap.c
+++ b/sis8300_mmap.c
@@ -55,12 +55,12 @@ static void sis8300_vm_close(struct vm_area_struct *vma) {
  * All pages that were mapped to userspace have already been allocated so this
  * should never happen.
  */
-static int sis8300_vm_fault(struct vm_area_struct *vma, struct vm_fault *vmfm) {
-    sis8300_usr *sisusr = vma->vm_private_data;
+static int sis8300_vm_fault(struct vm_fault *vmf) {
+    sis8300_usr *sisusr = vmf->vma->vm_private_data;
     sis8300_dev *sisdevice = sisusr->sisdevice;
     
     printk(KERN_INFO "%s: fault mapping: uaddr=0x%lx\n", 
-            sisdevice->name, vma->vm_start);
+            sisdevice->name, vmf->vma->vm_start);
     
     /* Disallow faulting in any additional pages; if NULL the kernel would
     allocate 'normal', non-contiguous memory. */
diff --git a/sis8300_read.c b/sis8300_read.c
index 46c01eb..26aa06c 100644
--- a/sis8300_read.c
+++ b/sis8300_read.c
@@ -3,7 +3,7 @@
  */
 
 #include <linux/fs.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 
 #include "sis8300.h"
 #include "sis8300_defs.h"
diff --git a/sis8300_write.c b/sis8300_write.c
index 2713849..2a4c13f 100644
--- a/sis8300_write.c
+++ b/sis8300_write.c
@@ -3,7 +3,7 @@
  */
 
 #include <linux/fs.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 
 #include "sis8300.h"
 #include "sis8300_defs.h"
