From 91bbb5defce15f75b71cbf07fc7a612a11841fb9 Mon Sep 17 00:00:00 2001
From: Oliver Talevski <oliver.talevski@esss.se>
Date: Wed, 7 Nov 2018 11:25:22 +0100
Subject: [PATCH] Add license file

---
 COPYING | 8 ++++++++
 1 file changed, 8 insertions(+)
 create mode 100644 COPYING

diff --git a/COPYING b/COPYING
new file mode 100644
index 0000000..61e7ee6
--- /dev/null
+++ b/COPYING
@@ -0,0 +1,8 @@
+#
+#    Copyright and all other rights in this document are reserved by
+#    IOxOS Technologies SA. This documents contains proprietary information
+#    and is supplied on express condition that it may not be disclosed,
+#    reproduced in whole or in part, or used for any other purpose other
+#    than that for which it is supplies, without the written consent of
+#    IOxOS Technologies SA
+#
-- 
2.7.4

